src/unused_finder/graph.rs

```diff
@@ -0,0 +1,104 @@
+use std::{
+    collections::{HashMap, HashSet},
+    rc::Rc,
+};
+
+use super::{
+    node_visitor::{ExportKind, ImportedItem},
+    unused_finder_visitor_runner::ImportExportInfo,
+};
+
+#[derive(Debug, Clone, Default)]
+pub struct GraphFile {
+    pub is_used: bool,
+    // pub package_name: String,
+    pub file_path: String,
+    pub unused_exports: HashSet<ExportKind>,
+    pub import_export_info: ImportExportInfo,
+}
+
+impl GraphFile {
+    pub fn mark_item_as_used(&mut self, item: &ImportedItem) -> bool {
+        let item = ExportKind::from(item);
+        self.unused_exports.remove(&item)
+    }
+}
+
+pub type Edge = (String, String, ImportedItem);
+
+#[derive(Default, Debug)]
+pub struct Graph {
+    pub files: HashMap<String, Rc<GraphFile>>,
+    pub edges: HashSet<Edge>,
+}
+
+impl<'a> Graph {
+    pub fn bfs(&mut self, entries: Vec<String>) -> Vec<String> {
```

Comment:
`bfs_step` -> this is a single expansion fo the current frontier set into a new frontier, not a whole bfs